detection_rules/ecs.py

```diff
@@ -150,6 +151,31 @@ def flatten(schema):
     return flattened
 
 
+@cached
+def get_all_flattened_schema() -> dict:
```

Comment:
Can we gate this behind custom rule usage with the auto_gen field so that we dont load this when not used?

Reply:
It is already gated, `get_all_flattened_schema()` is only called from `auto_add_field` which only gets called if both there is an unknown field error and `auto_gen_schema_file` is set in the rules config. 

e.g.
```python
if validation_checks["stack"].error_msg == "Unknown field" and RULES_CONFIG.auto_gen_schema_file:
    # auto add the field and re-validate
    self.auto_add_field(validation_checks["stack"], data.index_or_dataview[0])

```